mltensor-explainer.md

```diff
@@ -0,0 +1,415 @@
+await mlContext.compute(graph3, {'input': imageAsArrayBuffer}, {'output': outputArrayBuffer3});
+```
+
+Using `MLTensor`s enables a programming model similar to [WebGPU's](https://www.w3.org/TR/webgpu/#programming-model). Tasks are posted to the ML context's [timeline](#timelines) and are executed as the ML context sees fit - so far as data dependencies are respected. In this example, the ML context should be working continuously from the `writeBuffer()` call until the work for the last `readBuffer()` completes. Better utilization of the ML context will result in significantly better throughput.
+
+```js
+// Proposed approach to queue tasks to the ML context timeline
+
+// Post a task to the ML context timeline to allocate and zero out a tensor,
+// then return to script.
+const imageAsMlTensor = await mlContext.createTensor({..., usage: MLTensorUsage.WRITE_TO});
+
+// Post a task to the ML context timeline to write to the tensor. Note that we do
+// not await completion of this write. The ML context will ensure any operations
+// which depend on the contents of `imageAsMlTensor` will queue behind this task.
+mlContext.writeBuffer(imageAsMlTensor, imageAsArrayBuffer);
+
+// Post a task to the ML context timeline to execute the graph. The ML context will
+// ensure this queues behind the write above.
+mlContext.dispatch(graph1, {'input': imageAsMlTensor}, {'output': outputMlTensor1});
+// Post another task. Since input tensors will not be modified by graph execution,
+// the ML context may choose to execute in parallel to the dispatch above.
+mlContext.dispatch(graph2, {'input': imageAsMlTensor}, {'output': outputMlTensor2});
+// Post another task, which may also execute in parallel.
+mlContext.dispatch(graph3, {'input': imageAsMlTensor}, {'output': outputMlTensor3});
+
+// Post tasks to read the output tensors. These tasks will queue behind the
+// respective dispatch() calls using each tensor.
+const outputs = await Promise.all([
+    outputMlTensor1,
+    outputMlTensor2,
+    outputMlTensor3
+  ].map((tensor) => { return mlContext.readBuffer(tensor); }));
+```
+
+Since the queueing mechanism respects data dependencies, chained inference allows an `MLTensor` to be passed as an output from one graph and then immediately as an input to the next. A collection of graphs and buffers may be repeatedly dispatched without the need for synchronization via script.
+
+```js
+// Computing the Nth Fibonacci number using chained inference 
+
+const builder = new MLGraphBuilder(mlContext);
+const fn1 = builder.input('F_n-1', {dataType: "int32", shape: [1]});
+const fn2 = builder.input('F_n-2', {dataType: "int32", shape: [1]});
+const add = builder.add(fn1, fn2);
+const graph = await builder.build({'F_n': add});
+
+const descriptor = {dataType: "int32", shape: [1], usage: MLTensorUsage.WRITE_TO};
+const tensors = await Promise.all([
+    mlContext.createTensor(descriptor),
+    mlContext.createTensor(descriptor),
+    mlContext.createTensor(descriptor)
```

Comment:
Ah yes, good catch! I tested this code to ensure it works, but that was before @bbernhar's [CL implementing buffer usages](https://crrev.com/c/5787041) merged

It's not quite as simple as making the output tensor readable, since `N` can be any arbitrary value in this example. Added the code to make this work, though!